test/cpp/end2end/client_lb_end2end_test.cc

```diff
@@ -798,6 +798,38 @@ TEST_F(ClientLbEnd2endTest, PickFirstUpdateSuperset) {
   EXPECT_EQ("pick_first", channel->GetLoadBalancingPolicyName());
 }
 
+TEST_F(ClientLbEnd2endTest, PickFirstUpdateToUnconnected) {
+  const int kNumServers = 4;
```

Comment:
There's not actually any need to create servers that we never need to start.  Instead, you can just push the result of `grpc_pick_unused_port_or_die()` onto `ports` when you want a server that won't respond.

In fact, it's not clear to me that we actually need more than one server here.  I think we could just start one server and have the resolver first send that server's address, then send `grpc_pick_unused_port_or_die()`, then send that same server's address again.

Reply:
I simplified by removing some of the servers, however I think that having two servers we increase the test coverage.

By handling the second resolution with an unavailable but later available server we test that the second resolution is installed and works once that server becomes available.  Sending a new resolution for server[0] again would work but it isn't testing the same thing since that resolution would overwrite the previous.